aggregator/src/main/java/com/vmware/flowgate/aggregator/scheduler/job/AggregatorService.java

```diff
@@ -137,6 +147,91 @@ public void mergeServerMapping() {
+         if(pduFromPowerIQ != null && !pdu.getId().equals(pduFromPowerIQ.getId())) {
+            HashMap<String,String> pduFromPowerIQExtraInfo = pduFromPowerIQ.getJustificationfields();
+            HashMap<String,String> pduExtraInfo = pdu.getJustificationfields();
+            pdu.setAssetSource(pdu.getAssetSource() + FlowgateConstant.SPILIT_FLAG + pduFromPowerIQ.getId());
+            if(pduExtraInfo == null || pduExtraInfo.isEmpty()) {
+               pdu.setJustificationfields(pduFromPowerIQExtraInfo);
+               restClient.saveAssets(pdu);
+               continue;
+            }
+            String pduInfo = pduFromPowerIQExtraInfo.get(FlowgateConstant.PDU);
+            if(pduInfo == null) {
+               continue;
+            }
+            String oldPduInfo = pduExtraInfo.get(FlowgateConstant.PDU);
+            if(oldPduInfo == null) {
+               pduExtraInfo.put(FlowgateConstant.PDU, pduInfo);
+               restClient.saveAssets(pdu);
+               continue;
+            }
+            Map<String,String> pduInfoMap = null;
+            Map<String,String> oldPduInfoMap = null;
+            try {
+               pduInfoMap = mapper.readValue(pduInfo, new TypeReference<Map<String,String>>() {});
+               oldPduInfoMap = mapper.readValue(oldPduInfo, new TypeReference<Map<String,String>>() {});
+            } catch (IOException e) {
+               logger.error("Format pdu extra info error");
+               continue;
+            }
+            oldPduInfoMap.put(FlowgateConstant.PDU_RATE_AMPS, pduInfoMap.get(FlowgateConstant.PDU_RATE_AMPS));
+            oldPduInfoMap.put(FlowgateConstant.PDU_MIN_RATE_POWER, pduInfoMap.get(FlowgateConstant.PDU_MIN_RATE_POWER));
+            oldPduInfoMap.put(FlowgateConstant.PDU_MAX_RATE_POWER, pduInfoMap.get(FlowgateConstant.PDU_MAX_RATE_POWER));
+            oldPduInfoMap.put(FlowgateConstant.PDU_MIN_RATE_VOLTS, pduInfoMap.get(FlowgateConstant.PDU_MIN_RATE_VOLTS));
+            oldPduInfoMap.put(FlowgateConstant.PDU_MAX_RATE_VOLTS, pduInfoMap.get(FlowgateConstant.PDU_MAX_RATE_VOLTS));
+            oldPduInfoMap.put(FlowgateConstant.PDU_OUTLETS_FROM_POWERIQ, pduInfoMap.get(FlowgateConstant.PDU_OUTLETS_FROM_POWERIQ));
+            oldPduInfoMap.put(FlowgateConstant.PDU_INLETS_FROM_POWERIQ, pduInfoMap.get(FlowgateConstant.PDU_INLETS_FROM_POWERIQ));
+            try {
+               String newPduInfo = mapper.writeValueAsString(oldPduInfoMap);
+               pduExtraInfo.put(FlowgateConstant.PDU, newPduInfo);
+               pdu.setJustificationfields(pduExtraInfo);
+            } catch (JsonProcessingException e) {
+               logger.error("Format pdu extra info error",e.getCause());
+            }
+            restClient.saveAssets(pdu);
+         }
+      }
+      logger.info("Finish aggregate pdu from PowerIQ to other systems");
+
+      for(Map.Entry<String, Asset> map : pdusOnlyFromPowerIQ.entrySet()) {
```

Comment:
This is not correct, you should only remove the asset that has been merged.

Reply:
which mean you should only delete an PDUfromPowerIQ only if you call restClient.SaveAssets(pdu)

Reply:
Done